### PR TITLE
[INLONG-3405][Manager] Support generic partition configuration for Hive sink

### LINKED ISSUE #3405: [Improve][Manager] Support generic partition configuration for Hive sink
### Description

Support generic hive partition configuration, currently only support primary partition and secondary partition configuration.

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by yunqingmoswu**
### Title Name: [INLONG-3405][Manager] Support generic partition configuration for Hive sink

Fixes #3405 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/pojo/sink/SinkPartitionFieldInfo.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+import lombok.Data;
+
+/**
+ * Sink partition field info
+ */
+@Data
+@ApiModel("Sink partition field info")
+public class SinkPartitionFieldInfo {
```
Comment (healchow) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/pojo/sink/SinkPartitionFieldInfo.java:29]: It is recommended to move it to `pojo/sink/hive` package and name it `HivePartitionField`.

File: inlong-manager/manager-web/sql/apache_inlong_manager.sql [hunk 2]
```diff
@@ -16,13 +16,16 @@
  */
 
 SET NAMES utf8mb4;
-SET FOREIGN_KEY_CHECKS = 0;
```
Comment (healchow) [inline inlong-manager/manager-web/sql/apache_inlong_manager.sql:19]: Please do not format this file.

File: inlong-manager/manager-dao/src/main/resources/mappers/StreamSinkFieldEntityMapper.xml [hunk 3]
```diff
@@ -46,8 +46,7 @@
         insert into stream_sink_field (id, inlong_group_id, inlong_stream_id,
                                        sink_id, sink_type, field_name,
                                        field_type, field_comment, source_field_name,
-                                       source_field_type, is_meta_field, field_format,
```
Comment (healchow) [inline inlong-manager/manager-dao/src/main/resources/mappers/StreamSinkFieldEntityMapper.xml:49]: Do not change the format.

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/pojo/sink/SinkPartitionFieldInfo.java (hunk 1 above)
Comment (yunqingmoswu) [reply inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/pojo/sink/SinkPartitionFieldInfo.java]: done

File: inlong-manager/manager-web/sql/apache_inlong_manager.sql (hunk 2 above)
Comment (yunqingmoswu) [reply inlong-manager/manager-web/sql/apache_inlong_manager.sql]: done

File: inlong-manager/manager-dao/src/main/resources/mappers/StreamSinkFieldEntityMapper.xml (hunk 3 above)
Comment (yunqingmoswu) [reply inlong-manager/manager-dao/src/main/resources/mappers/StreamSinkFieldEntityMapper.xml:49]: done

**Review by healchow — APPROVED** [review]
LGTM

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**